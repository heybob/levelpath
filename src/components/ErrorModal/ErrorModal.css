.modal-background {
    display: flex;
    width: 100%;
    height: 100vh;
    background-color: rgb(0,0,0,0.25);
    z-index: 100;
    position: absolute;
    top: 0;
    justify-content: center;
    align-items: center;
    flex-direction: column;

}
.modal-error {
    background: white;
    border-radius: 1rem;
    box-shadow: 2px 1px 10px rgb(0,0,0,0.45);
}
.modal-error header {
    border-bottom: 1px solid lightgray;
    padding: 1rem;
    background-color: rgb(179, 3, 3);
    border-radius: 1rem 1rem 0 0;
    color: white;
    
}

.modal-error footer {
    border-top: 1px solid lightgray;
    padding: 0.5rem;
}

.modal-error section {
    padding: 1rem;
    font-weight: bold;
}
.modal-error p {
    margin: 0.5rem;
}