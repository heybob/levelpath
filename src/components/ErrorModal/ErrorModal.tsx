import { useContext,useCallback } from "react"
import { WikiContext } from "../../contexts/WikiContext"
import './ErrorModal.css'
import { closeErrorModal } from "../App/actions";
import { Text } from "../../shared/constants";
import { Button } from "../Button/Button";

export function ErrorModal() {
    const {dispatch} = useContext(WikiContext);
    const handleClick = useCallback(() => (dispatch(closeErrorModal())), [dispatch]);
    
    return (
    <div className="modal-background">
        <div className="modal-error">
            <header>
                <h1>Error</h1>
            </header>
            <section>
                <p>{Text.errorMessage1}</p>
                <p>{Text.errorMessage2}</p>
            </section>
            <footer>
                <Button name={Text.close} clickHandler={handleClick} />
            </footer>
        </div>
    </div>    
    )
}