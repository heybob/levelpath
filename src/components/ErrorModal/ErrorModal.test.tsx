import renderer from 'react-test-renderer';
import { ErrorModal } from "./ErrorModal";

describe('Testing Loading Indicator Component', () => {
  it('It should render loading indicator', () => {
      const tree = renderer.create(<ErrorModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
