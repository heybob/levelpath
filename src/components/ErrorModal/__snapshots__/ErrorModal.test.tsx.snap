// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading Indicator Component It should render loading indicator 1`] = `
<div
  className="modal-background"
>
  <div
    className="modal-error"
  >
    <header>
      <h1>
        Error
      </h1>
    </header>
    <section>
      <p>
        We're sorry, an error happened while getting the data.
      </p>
      <p>
        Please try again.
      </p>
    </section>
    <footer>
      <button
        disabled={false}
        onClick={[Function]}
      >
        Close
      </button>
    </footer>
  </div>
</div>
`;
